src/cosign/mod.rs

```diff
@@ -229,6 +236,19 @@ where
     }
 }
 
+/// Verifies the blob using the signature, and the certificate.
```

Comment:
I think the inline documentation should be more clear:

```suggestion
/// Verifies the signature produced by cosign when signing the given blob via the `cosign sign-blob` command
///
/// The parameters:
/// * `cert`: a PEM encoded x509 certificate that contains the public key used to verify the signature
/// * `signature`: the base64 encoded signature of the blob that has to be verified
/// * `blob`: the contents of the blob
///
/// This function returns `Ok())` when the given signature has been verified, otherwise returns an `Err`.
```

As a result of that, I think we can rename the `base64_signature` to be just `signature`.